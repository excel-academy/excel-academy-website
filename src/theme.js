import { themeColor } from '../site-config';

const purples = ['#7b79b0', '#6f6da5', '#eeeefc', '#1c2070'];
const greens = ['#d2eced', '#132c2d'];
const fontSizes = [12, 14, 16, 18, 20, 24, 30, 38, 62];
const space = [0, '2rem', '3rem', '4rem', '5rem', '7rem', '32rem', '40rem', '48rem', '56rem', '64rem'];
export const breakpoints = {
  sm: 0,
  mobile: '30em',
  tablet: '46em',
  desktop: '61.25em',
  wide: '80em',
  extrawide: '105em',
};

export default {
  space,
  fontSizes,
  breakpoints,
  maxWidths: ['48em', '60em', '80em'],
  colors: {
    theme: themeColor,
    purples,
    greens,
  },
  buttons: {
    primary: {
      color: '#fff',
      backgroundColor: themeColor,
      fontSize: fontSizes[4],
      borderColor: '#fff',
      paddingRight: space[4],
      paddingLeft: space[4],
      '&:hover': {
        backgroundColor: '#000',
      },
    },
    outline: {
      color: themeColor,
      backgroundColor: 'transparent',
      boxShadow: 'inset 0 0 2px',
    },
  },
};
