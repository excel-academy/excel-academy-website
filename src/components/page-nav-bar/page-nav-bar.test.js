import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from '../../theme';
import PageNavBar from './page-nav-bar';

describe('PageNavBar', () => {
  it('default renders correctly', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <PageNavBar navItems={[
          { id: 'test', title: 'Test' },
          { id: 'test2', title: 'Test 2' },
        ]}
        />
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
