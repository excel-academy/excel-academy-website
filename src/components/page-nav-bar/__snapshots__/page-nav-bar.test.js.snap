// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNavBar default renders correctly 1`] = `
.c0 {
  box-sizing: border-box;
  padding-left: 4rem;
  padding-right: 4rem;
  background-color: #d2eced;
  position: -webkit-sticky;
  position: sticky;
  z-index: 11;
  top: 0;
  bottom: 0;
  display: none;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  box-sizing: border-box;
  padding-top: 1em;
  padding-bottom: 1em;
  font-size: 1em;
  color: #132c2d;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  display: inline-block;
}

.c2:hover {
  color: #4bb2b6;
}

@media (min-width:46em) {
  .c0 {
    display: block;
  }
}

<nav
  className="c0"
>
  <ul
    className="c1"
  >
    <li>
      <a
        className="c2"
        color="greens.1"
        fontSize="1em"
        href="#test"
      >
        Test
      </a>
    </li>
    <li>
      <a
        className="c2"
        color="greens.1"
        fontSize="1em"
        href="#test2"
      >
        Test 2
      </a>
    </li>
  </ul>
</nav>
`;
